include/libdnf/rpm/package.hpp

```diff
@@ -38,280 +38,439 @@ class Goal;
+    /// @since 5.0
+    //
+    // TODO(dmach): add overridden method that allows specifying checksum type?
+    // TODO(dmach): repodata can contain multiple checksums - how are we going to handle these?
+    // @replaces dnf:dnf/package.py:attribute:Package.chksum
+    // @replaces libdnf:libdnf/hy-package-private.hpp:function:dnf_package_get_chksum(DnfPackage *pkg, int *type)
     Checksum get_checksum() const;
 
-    /// @replaces dnf:dnf/package.py:attribute:Package.hdr_chksum
-    /// @replaces libdnf:libdnf/hy-package-private.hpp:function:dnf_package_get_hdr_chksum(DnfPackage *pkg, int *type)
+    /// @return Checksum object representing RPM package header checksum and its type
+    /// @since 5.0
+    //
+    // TODO(dmach): add overridden method that allows specifying checksum type?
+    // TODO(dmach): repodata can contain multiple checksums - how are we going to handle these?
+    // TODO(dmach): does it come only from rpmdb or file? (I don't think it is part of repodata)
+    // @replaces dnf:dnf/package.py:attribute:Package.hdr_chksum
+    // @replaces libdnf:libdnf/hy-package-private.hpp:function:dnf_package_get_hdr_chksum(DnfPackage *pkg, int *type)
     Checksum get_hdr_checksum() const;
 
-    /// TODO get_changelogs - requires changelog
-    /// @replaces dnf:dnf/package.py:attribute:Package.changelogs
-    /// @replaces libdnf:libdnf/hy-package-private.hpp:function:dnf_package_get_changelogs(DnfPackage * pkg)
-    // void get_changelogs() const;
+    /// @return RPM package header end (header it is located between file position 0 and the hdr_end)
+    /// @since 5.0
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.hdr_end
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_hdr_end(DnfPackage * pkg)
+    unsigned long long get_hdr_end() const;
 
-    // SYSTEM
+    /// @return Number of media (usually a CD/DVD disc) on which the package is located
+    /// @since 5.0
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.medianr
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_medianr(DnfPackage * pkg)
+    unsigned long long get_media_number() const;
 
-    /// @replaces libdnf:libdnf/hy-package.h:function:dnf_package_installed(DnfPackage * pkg)
-    /// @replaces dnf:dnf/package.py:attribute:Package.installed
+    // ===== SYSTEM =====
+
+    /// @return RPM package location (file path) on the file system:
+    ///         * If the package is a local package, return its path
+    ///         * If the package is a remote package downloaded to cache, return path to the cache
+    ///         * If the package is a remote package that hasn't been downloaded yet, return path to the cache
+    /// @since 5.0
+    //
+    // @replaces dnf:dnf/package.py:attribute:Package.localPkg
+    // @replaces libdnf:libdnf/hy-package.h:function:dnf_package_get_local_baseurl(DnfPackage * pkg)
+    std::string get_local_path() const;
```

Comment:
It can somehow return an empty string, but it is not mentioned in description.

Reply:
What do you think to rename it to `get_local_package_path`?

Reply:
I don't think it can return an empty string. If it is possible, it deserves documentation and a test case.
I'm not sure about `get_local_package_path`. There's extra `package` in it, because it's part of the Package class. On the other hand, we have `get_package_size` too. Need to think about it.